source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -66,49 +67,58 @@ bool hasMatchingIpVersion(const Address::IpVersion& ip_version,
 } // namespace
 
 std::vector<Address::InstanceConstSharedPtr> HappyEyeballsConnectionProvider::sortAddresses(
-    const std::vector<Address::InstanceConstSharedPtr>& in) {
+    const std::shared_ptr<const std::vector<Address::InstanceConstSharedPtr>>& in) {
+  if (in == nullptr) {
```

Comment:
Interesting. I was under the impression that happy-eyeballs always needs a list of addresses, so this couldn't happen.
If this is just a circuit-breaker then why not also have `|| (in->size() == 0)` there as well?

Reply:
If you leave in the early-exit you should make sure it is tested.

I'm not certain we should change this interface at all though; maybe just put the `*` in at the call-site.

Reply:
Good idea to pass *address_list_ptr. With that, I removed the changes in the happy eyeball files.